src/pretix/plugins/sendmail/models.py

```diff
@@ -155,12 +155,13 @@ def send(self):
                     positions = [p for p in positions if p.subevent_id == self.subevent_id]
 
                 for p in positions:
-                    email_ctx = get_email_context(event=e, order=o, position_or_address=ia, position=p)
                     try:
                         if p.attendee_email and (p.attendee_email != o.email or not o_sent):
+                            email_ctx = get_email_context(event=e, order=o, position_or_address=p, position=p)
```

Comment:
Looking at the definition of `get_email_context`, if `position_or_address` is the same as `position`, we can omit it completely. So we might simplify – although I can see the benefit of clearly stating `position_or_address` here as to make the differentiation to line 164 more clear. I am undecided.

```suggestion
                            email_ctx = get_email_context(event=e, order=o, position=p)
```

Reply:
I made another change to actually *never* set ``position_or_address`` but always rely on the implementation inside ``get_email_context``. That feels much safer against future confusions!

Reply:
Passing the `invoice_address` in this case is just an optimization to not have it „calculated“ on every invocation of `get_email_context`, right? Because otherwise we could leave the `invoice_address=ia` parameter out as well.

Reply:
>Passing the invoice_address in this case is just an optimization to not have it „calculated“ on every invocation of get_email_context, right? 
Correct